File: c programming/2Darray/spiral.c
#include<stdio.h>
int main()
{
    int arr[5][6]={{1,2,3,4,5,6},{7,8,9,10,11,12},{13,14,15,16,17,18},{19,20,21,22,23,24},{25,26,27,28,29,30}};
    int minr=0;
    int maxc=5;
    int maxr=4;
    int minc=0;
    int count =0;
    int te=(5)*(6);
    while(count<te)
    {
        //minimum row
        for(int i=minc;i<=maxc;i++)
        {
            printf("%d ",arr[minr][i]);
            count++;
        }
        minr++;
        if(count>=te)
        break;
        //maximum column
        for(int j=minr;j<=maxr;j++)
        {
            printf("%d ",arr[j][maxc]);
            count++;
        }
        maxc--;
        if(count>=te)
        break;
        //maximum row(reverse direction)
        for(int k=maxc;k>=minc;k--)
        {
            printf("%d ",arr[maxr][k]);
            count++;
        }
        maxr--;
        if(count>=te)
        break;
        // minimum row(reverse order)
        for(int m=maxr;m>=minr;m--)
        {
            printf("%d ",arr[m][minc]);
            count++;
        }
        minc++;
        if(count>=te)
        break;
    }
}

File: jAVA/twoD_array/spiral.java


public class spiral {
    public static void main(String[] args) {
        int arr[][]={{1,2,3},{4,5,6},{7,8,9}};
        int minr=0;
        int minc=0;
        int maxr=arr.length-1;
        int maxc=arr[0].length-1;
        int te=arr.length*arr[0].length;
        int count=0;
        while(minr<=maxr && minc<=maxc)
        {
            //first row
            for(int i=minc;i<=maxc && count<te;i++)
            {
                System.out.print(arr[minr][i]+" ");
                count++;
                if(count%3==0)
                {
                System.out.print("\n");
                }
            }
            
            minr++;
            for(int j=minr;j<=maxr && count<te;j++)
            {
                System.out.print(arr[j][maxc]+" ");
                count++;
                if(count%3==0)
                {
                System.out.print("\n");
                }
            }
           // System.out.print("\n");
            maxc--;
            for(int k=maxc;k>=minc && count<te;k--)
            {
                System.out.print(arr[maxr][k]+" ");
                count++;
                if(count%3==0)
                {
                System.out.print("\n");
                }
            }
            //System.out.print("\n");
            maxr--;
            for(int m=maxr;m>=minr && count<te;m--)
            {
                System.out.print(arr[m][minc]+" ");
                count++;
                if(count%3==0)
                {
                System.out.print("\n");
                }
            }
           // System.out.print("\n");
            minc++;
        }
    }
}
